core/tests/protractor_utils/general.js

```diff
@@ -128,16 +134,25 @@ var moveToEditor = async function() {
 };
 
 var expect404Error = async function() {
-  expect(await element(by.css('.protractor-test-error-container')).getText()).
+  var protractorTestErrorContainer =
+    element(by.css('.protractor-test-error-container'));
+  await waitFor.visibilityOf(
+    protractorTestErrorContainer,
+    'Protractor test error container taking too long to appear');
+  expect(await protractorTestErrorContainer.getText()).
     toMatch('Error 404');
 };
 
 // Checks no untranslated values are shown in the page.
 var ensurePageHasNoTranslationIds = async function() {
   // The use of the InnerHTML is hacky, but is faster than checking each
   // individual component that contains text.
-  var promiseValue = await element(by.css(
-    '.oppia-base-container')).getAttribute('innerHTML');
+  var oppiaBaseContainer = element(by.css(
+    '.oppia-base-container'));
+  waitFor.visibilityOf(
```

Comment:
Do we need an `await` here?

Reply:
I think we will need an await here because the element finder will need to wait for finding the container before getting its attribute.

Reply:
Yes here as well.